src/PhoneNumberFormatService.php

```diff
@@ -0,0 +1,51 @@
+<?php
+declare(strict_types=1);
+
+namespace DR\Internationalization;
+
+use DR\Internationalization\PhoneNumber\PhoneNumberFormatOptions;
+use InvalidArgumentException;
+use libphonenumber\NumberParseException;
+use libphonenumber\PhoneNumber;
+use libphonenumber\PhoneNumberUtil;
+use RuntimeException;
+
+class PhoneNumberFormatService
+{
+    private PhoneNumberFormatOptions $defaultOptions;
+    private ?PhoneNumberUtil $phoneNumberUtil = null;
+
+    public function __construct(PhoneNumberFormatOptions $phoneNumberOptions)
+    {
+        $this->defaultOptions = $phoneNumberOptions;
+    }
+
+    public function format(string $phoneNumber, PhoneNumberFormatOptions $options = null): string
+    {
+        $countryCode = $options?->getDefaultCountryCode() ?? $this->defaultOptions->getDefaultCountryCode();
+        $format      = $options?->getFormat() ?? $this->defaultOptions->getFormat();
+        if ($format === null) {
+            throw new InvalidArgumentException('PhoneNumberOptions: unable to format phoneNumber without a given format');
+        }
+
+        $this->phoneNumberUtil ??= PhoneNumberUtil::getInstance();
+
+        try {
+            $parsedNumber = $this->phoneNumberUtil->parse($phoneNumber, $countryCode);
+        } catch (NumberParseException $e) {
+            throw new RuntimeException("Unable to parse phoneNumber: " . $phoneNumber, 0, $e);
```

Comment:
Isn't this invalid argument exception

Reply:
How to avoid triggering this error on user input? Should we add some isValid method to this library?

Reply:
The second question also resolved?

Reply:
Missed that comment when resolving this.
Validating phoneNumbers is for now out of scope, but is something I want to add to this library in the future (along with address validation), using separate validationService classes.